app/src/main/java/app/gamenative/utils/StorageUtils.kt

```
@@ -76,6 +80,56 @@ object StorageUtils {
+            try {
+                for (volume in sm.storageVolumes) {
+                    if (volume.state != Environment.MEDIA_MOUNTED) continue
+
+                    val volumeDir = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
+                        volume.directory
+                    } else {
+                        // Use reflection for older APIs (26-29) if getExternalFilesDirs missed it
+                        try {
+                            val getPath = volume.javaClass.getMethod("getPath")
+                            (getPath.invoke(volume) as? String)?.let { File(it) }
+                        } catch (re: Exception) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)